src/huggingface_hub/inference/_client.py

```diff
@@ -1384,15 +1500,40 @@ def object_detection(
         [ObjectDetectionOutputElement(score=0.9486683011054993, label='person', box=ObjectDetectionBoundingBox(xmin=59, ymin=39, xmax=420, ymax=510)), ...]
         ```
         """
-        # detect objects
-        response = self.post(data=image, model=model, task="object-detection")
+        parameters = {
+            "threshold": threshold,
+        }
+        if all(parameter is None for parameter in parameters.values()):
+            # if no parameters are provided, the image can be raw bytes, an image file, or URL to an online image
+            data = image
+            payload: Optional[Dict[str, Any]] = None
+        else:
+            # if parameters are provided, the image needs to be a base64-encoded string
+            data = None
+            payload = {"inputs": _b64_encode(image)}
+            for key, value in parameters.items():
+                if value is not None:
+                    payload.setdefault("parameters", {})[key] = value
```

Comment:
(out of scope for this PR) I feel that we should factorize the logic to handle `inputs` + `parameters` i.e. rules like "base64 encode only if at least 1 parameter", "provide parameters if at least 1 parameter", "provide only not none parameters".

Reply:
agree!